### ISSUE TITLE
tracing 0.1 fails to compile with `log` feature enabled

### ISSUE THREAD

**Issue by anp** [tokio-rs/tracing#137]
Hello! I am very excited to switch from tokio-trace 0.1 to tracing 0.1! Sadly I appear to be unable to compile the new release with the `log` feature enabled.

### Platform

Windows 64-bit

### Crates

Cargo.toml:

```toml
tracing = { version = "0.1", features = ["log"] }
```

Compiler error:

```
   Compiling tracing v0.1.0
       Fresh syn v0.15.39
     Running `rustc --crate-name tracing C:\Users\adam\.cargo\registry\src\github.com-1ecc6299db9ec823\tracing-0.1.0\src\lib.rs --color always --crate-type lib --emit=dep-info,metadata,link -C debuginfo=2 --cfg "feature=\"log\"" -C metadata=22d8d61eee879087 -C extra-filename=-22d8d61eee879087 --out-dir C:\Users\adam\Documents\GitHub\moxie\target\debug\deps -L dependency=C:\Users\adam\Documents\GitHub\moxie\target\debug\deps --extern cfg_if=C:\Users\adam\Documents\GitHub\moxie\target\debug\deps\libcfg_if-43544de485dae8ac.rlib --extern log=C:\Users\adam\Documents\GitHub\moxie\target\debug\deps\liblog-78ad9be729e51666.rlib --extern tracing_core=C:\Users\adam\Documents\GitHub\moxie\target\debug\deps\libtracing_core-db15944ba314627e.rlib --cap-lints allow`
       Fresh proc-macro-hack v0.5.7
       Fresh topo-macro v0.8.0 (C:\Users\adam\Documents\GitHub\moxie\topo\macro)
       Fresh futures-select-macro-preview v0.3.0-alpha.16
   Compiling topo v0.8.0 (C:\Users\adam\Documents\GitHub\moxie\topo)
     Running `rustc --edition=2018 --crate-name topo topo\src\lib.rs --color always --crate-type lib --emit=dep-info,metadata,link -C debuginfo=2 -C metadata=f800424da5f8c4bc -C extra-filename=-f800424da5f8c4bc --out-dir C:\Users\adam\Documents\GitHub\moxie\target\debug\deps -C incremental=C:\Users\adam\Documents\GitHub\moxie\target\debug\incremental -L dependency=C:\Users\adam\Documents\GitHub\moxie\target\debug\deps --extern owning_ref=C:\Users\adam\Documents\GitHub\moxie\target\debug\deps\libowning_ref-4e7c10f045132e25.rlib --extern scopeguard=C:\Users\adam\Documents\GitHub\moxie\target\debug\deps\libscopeguard-6b874d16a23e70b9.rlib --extern topo_macro=C:\Users\adam\Documents\GitHub\moxie\target\debug\deps\topo_macro-584f7209c63cec74.dll`
error[E0616]: field `level` of struct `tracing_core::Metadata` is private
   --> C:\Users\adam\.cargo\registry\src\github.com-1ecc6299db9ec823\tracing-0.1.0\src\span.rs:678:34
    |
678 |             .level(level_to_log!(self.meta.level))
    |                                  ^^^^^^^^^^-----
    |                                            |
    |                                            help: a method `level` also exists, call it with parentheses: `level()`

error[E0616]: field `target` of struct `tracing_core::Metadata` is private   --> C:\Users\adam\.cargo\registry\src\github.com-1ecc6299db9ec823\tracing-0.1.0\src\span.rs:679:21
    |
679 |             .target(self.meta.target)
    |                     ^^^^^^^^^^------
    |                               |
    |                               help: a method `target` also exists, call it with parentheses: `target()`

error[E0616]: field `module_path` of struct `tracing_core::Metadata` is private
   --> C:\Users\adam\.cargo\registry\src\github.com-1ecc6299db9ec823\tracing-0.1.0\src\span.rs:685:34
    |
685 |                     .module_path(self.meta.module_path)
    |                                  ^^^^^^^^^^-----------
    |                                            |
    |                                            help: a method `module_path` also exists, call it with parentheses: `module_path()`

error[E0616]: field `file` of struct `tracing_core::Metadata` is private
   --> C:\Users\adam\.cargo\registry\src\github.com-1ecc6299db9ec823\tracing-0.1.0\src\span.rs:686:27
    |
686 |                     .file(self.meta.file)
    |                           ^^^^^^^^^^----
    |                                     |
    |                                     help: a method `file` also exists, call it with parentheses: `file()`

error[E0616]: field `line` of struct `tracing_core::Metadata` is private
   --> C:\Users\adam\.cargo\registry\src\github.com-1ecc6299db9ec823\tracing-0.1.0\src\span.rs:687:27
    |
687 |                     .line(self.meta.line)
    |                           ^^^^^^^^^^----
    |                                     |
    |                                     help: a method `line` also exists, call it with parentheses: `line()`

error: aborting due to 5 previous errors

For more information about this error, try `rustc --explain E0616`.
error: Could not compile `tracing`.

Caused by:
  process didn't exit successfully: `rustc --crate-name tracing C:\Users\adam\.cargo\registry\src\github.com-1ecc6299db9ec823\tracing-0.1.0\src\lib.rs --color always --crate-type lib --emit=dep-info,metadata,link -C debuginfo=2 --cfg "feature=\"log\"" -C metadata=22d8d61eee879087 -C extra-filename=-22d8d61eee879087 --out-dir C:\Users\adam\Documents\GitHub\moxie\target\debug\deps -L dependency=C:\Users\adam\Documents\GitHub\moxie\target\debug\deps --extern cfg_if=C:\Users\adam\Documents\GitHub\moxie\target\debug\deps\libcfg_if-43544de485dae8ac.rlib --extern log=C:\Users\adam\Documents\GitHub\moxie\target\debug\deps\liblog-78ad9be729e51666.rlib --extern tracing_core=C:\Users\adam\Documents\GitHub\moxie\target\debug\deps\libtracing_core-db15944ba314627e.rlib --cap-lints allow` (exit code: 1)
```

Cargo.lock:

<details>
# This file is automatically @generated by Cargo.
# It is not intended for manual editing.
[[package]]
name = "cfg-if"
version = "0.1.9"
source = "registry+https://github.com/rust-lang/crates.io-index"

[[package]]
name = "lazy_static"
version = "1.3.0"
source = "registry+https://github.com/rust-lang/crates.io-index"

[[package]]
name = "log"
version = "0.4.6"
source = "registry+https://github.com/rust-lang/crates.io-index"
dependencies = [
 "cfg-if 0.1.9 (registry+https://github.com/rust-lang/crates.io-index)",
]

[[package]]
name = "test-tracing"
version = "0.1.0"
dependencies = [
 "tracing 0.1.0 (registry+https://github.com/rust-lang/crates.io-index)",
]

[[package]]
name = "tracing"
version = "0.1.0"
source = "registry+https://github.com/rust-lang/crates.io-index"
dependencies = [
 "cfg-if 0.1.9 (registry+https://github.com/rust-lang/crates.io-index)",
 "log 0.4.6 (registry+https://github.com/rust-lang/crates.io-index)",
 "tracing-core 0.1.0 (registry+https://github.com/rust-lang/crates.io-index)",
]

[[package]]
name = "tracing-core"
version = "0.1.0"
source = "registry+https://github.com/rust-lang/crates.io-index"
dependencies = [
 "lazy_static 1.3.0 (registry+https://github.com/rust-lang/crates.io-index)",
]

[metadata]
"checksum cfg-if 0.1.9 (registry+https://github.com/rust-lang/crates.io-index)" = "b486ce3ccf7ffd79fdeb678eac06a9e6c09fc88d33836340becb8fffe87c5e33"
"checksum lazy_static 1.3.0 (registry+https://github.com/rust-lang/crates.io-index)" = "bc5729f27f159ddd61f4df6228e827e86643d4d3e7c32183cb30a1c08f604a14"
"checksum log 0.4.6 (registry+https://github.com/rust-lang/crates.io-index)" = "c84ec4b527950aa83a329754b01dbe3f58361d1c5efacd1f6d68c494d08a17c6"
"checksum tracing 0.1.0 (registry+https://github.com/rust-lang/crates.io-index)" = "ff6ff2664f0b91870d297cd3e51be6980057273998844578fb392487cb93cfc1"
"checksum tracing-core 0.1.0 (registry+https://github.com/rust-lang/crates.io-index)" = "7d5f34cae5ddd9e2dab25f17ff18acec0832b35c8d0ef9a9528737da74370b4b"

</details>

**Comment by hawkw** [comment]
Yeah, this is my fault --- the `log` feature's integration tests were misplaced in the repository move, and then we made some changes that broke that feature by mistake. I'm working on fixing it in #131, but have hit some odd test failures.

Once this is fixed I'll get `tracing` 0.1.1 out ASAP.

**Comment by hawkw** [comment]
> the `log` feature's integration tests were misplaced in the repository move

If you're curious, the gory details are basically just that the `log` feature has to be tested in a separate crate, since it's a feature flag that's off by default, and I forgot to have CI test the separate test crate when moving over the CI configs from the tokio repo. Sadly, not that exciting.

**Comment by hawkw** [comment]
Ah, I've [figured out](https://github.com/tokio-rs/tracing/pull/131/commits/97f6ff1f759d43476cec86064621cb803b0646be) why the PR that fixes the `log` feature wasn't passing CI --- should get the fix merged shortly!

**Comment by hawkw** [comment]
@anp Just wanted to let you know that we've published [`tracing` v0.1.1](https://crates.io/crates/tracing/0.1.1) with this fix!
